modules/piezo-processor/main.py

```
@@ -205,16 +280,22 @@ def read_records(self):
-                inner = cbor2.loads(record["data"])
+                data_bytes = _read_raw_record(self._file)
+                if data_bytes is None:
+                    self._last_pos = self._file.tell()
+                    continue  # empty placeholder record
+                inner = cbor2.loads(data_bytes)
+                self._last_pos = self._file.tell()
                 yield inner
             except EOFError:
                 # No new data yet — poll
                 time.sleep(0.01)
             except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)